doc/source/tutorial/stats/outliers.md

```diff
@@ -0,0 +1,521 @@
+np.mean(x[2:-2])
+```
+
+```{code-cell} ipython3
+# Remove `ceil(0.1 * 22)` (2) observations from each tail
+np.mean(x[3:-3])
+```
+
+What if instead of removing a proportion, we want to remove data below and above certain threshholds. It turns out that we would need to use `scipy.stats.tmean` or (the similarly-named, but curiously relegated to a separate namespace) `scipy.stats.mstats.trimmed_mean`, instead.
+
+```{code-cell} ipython3
+stats.tmean(x, limits=(2, 19), inclusive=(True, True))
+```
+
+```{code-cell} ipython3
+stats.mstats.trimmed_mean(x, limits=(2, 19), relative=False, inclusive=(True, True))
+```
+
+What if we want to remove data outside a "confidence interval" estimate. Do any of these functions - or perhaps `scipy.stats.trim` followed by `np.mean` - offer anything to facilitate that?
+
+No, not really.
+
+So at the time of writing, SciPy offers three or four ways of performing some operations related to trimming, but none are able to perform *all* operations we might reasonably expect them to. In the spirit of the Zen of Python, we suggest the new "one-- and preferably only one --obvious way to do it.": manually, in one or two lines of code, using more familiar, general-purpose functions. Whether calculating a trimmed or winsorized statistic, there will be three steps:
+
+1. Identify the outlier threshold.
+2. Trim or winsorize.
+3. Compute the statistic.
+
++++
+
+# Identifying the threshold
+
+SciPy now offers a `quantile` function with several methods useful for identifying thresholds beyond which data are to be trimmed or winsorized. The most common thresholds are those corresponding with a certain *percentage* of the data in each tail. To set the threshold at $p=10\%$ of the data in each tail, rounding the number of points to trim *down*:
+
+```{code-cell} ipython3
+p = 0.1
+stats.quantile(x, [p, 1-p], method='winsor_less')
+```
+
+Rounding the number of points to trim *up*:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='winsor_more')
+```
+
+Rounding the number of points to trim to the nearest integer:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='winsor_round')
+```
+
+Note that these methods are all subtlely different from estimating the 10th and 90th percentiles of the data. Using the default method:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p])
+```
+
+The Harrel-Davis method:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='harrell-davis')
+```
+
+Note that `quantile` is fully vectorized. Suppose we have two rows of data, and we wish to set different thresholds for each row.
+
+```{code-cell} ipython3
+x2d = np.stack([x, x])
+p2d = [[p  , 1 - p  ],  # thresholds for zeroth row
+       [2*p, 1 - 2*p]]  # thresholds for next row
+stats.quantile(x2d, p2d, method='winsor_less', axis=-1)
+```
+
+## Missing Data
+
++++
+
+If there are values to be ignored, use an [MArray](https://mdhaber.github.io/marray/tutorial.html) to mask them out. 
```

Comment:
I should probably include a short section motivating the transition from NumPy masked arrays to MArrays here, too?

Reply:
Not here.